### ISSUE TITLE
Hitting enter in a form field within a <form> calls onRemove handler for tags

### ISSUE THREAD

**Issue by chaddjohnson** [Shopify/polaris#235]
# Issue summary

I have a form with `TextField` and `Tag` components within. The `Tag` components have `onRemove` handlers. If I focus on and hit enter on one of the `TextField` components, the `onRemove` handler for one of the `Tag` elements is called. 

I consider this to be a bug.

A user would expect the form to submit, but instead, their tags would be removed each time they hit enter in frustration.

Note that the handler is *not* called if the components are not surrounded by a `<form>` tag.

## Expected behavior

The `onRemove` handler for a `Tag` component should only ever be called when the `X` button for that tag is clicked.

## Actual behavior

The `onRemove` handler for `Tag` components is being called when it should be.

## Steps to reproduce the problem

1. Clone the repository at https://github.com/chaddjohnson/polaris.
1. Run the webpack example and open it in your browser.
1. Visit any text field in the "Form" section.
1. Hit enter.
1. Notice in the console that "handleRemoveTag() called" is printed.

## Specifications

- Polaris version: 1.8.3
- React version: 16
- Browser: Chrome
- Device: MacBook
- Operating System: MacOS High Sierra

---

⚓️ We’re not accepting pull requests at this time<br />
🗒 This repo is for reporting issues and feature requests only

**Comment by ridem** [comment]
I've noticed this issue as well.

We can't send PR, so here's the thing:

The Tag component's delete element is actually a `<button>` for which the type hasn't been set. So by default, it's a `button[type=submit]`, which in this case is placed before the actual "Submit" button of the form.
When you press enter, all onClick events of `button[type=submit]` are fired before the `<form>`'s own `onSubmit` event, hence your bug.

**File:**
https://github.com/Shopify/polaris/blob/master/src/components/Tag/Tag.tsx#L15

**Fix:**
Add `type="button"` to the `<button>` in that file

**Comment by chaddjohnson** [comment]
Hi @chloerice, any chance we can get the fix @ridem suggested in place for this issue?

**Comment by chaddjohnson** [comment]
I was able to use @ridem's solution and workaround the problem without modifying Polaris:

```javascript
componentDidUpdate(prevProps, prevState) {
    document.querySelectorAll('.Polaris-Tag__Button').forEach(button => {
        button.setAttribute('type', 'button');
    });
}
```

**Comment by chloerice** [comment]
Hey @chaddjohnson the fix has been made and will come out with the next patch release, thanks for raising this!

**Comment by chaddjohnson** [comment]
Thank you Chloe! :)

**Comment by dfmcphee** [comment]
This fix was released in v1.11.0.
